php-checks/src/main/java/org/sonar/php/checks/ClassNameCheck.java

```diff
@@ -61,4 +67,15 @@ public void visitNode(Tree tree) {
     }
   }
 
+  /**
+   * Skip generated Yii DB Migration classes as they have a specific naming convention.
+   */
+  private static boolean skipYiiDbMigrationClass(ClassDeclarationTreeImpl tree) {
+    ClassSymbol superClass = tree.symbol().superClass().orElse(null);
+    if (superClass == null) {
+      return false;
+    }
+    String qualifiedName = superClass.qualifiedName().toString();
+    return "yii\\db\\migration".equals(qualifiedName);
```

Comment:
Not mentioned in the ticket, but while we are at it let's add `\yii\mongodb\Migration` see https://github.com/search?q=%2Fextends.*yii%5C%5Cmongodb%5C%5CMigration%2F+language%3APHP+&type=code